Assets/Scripts/GridBlock.cs

```diff
@@ -10,23 +10,74 @@ public class GridBlock
 
     int y;
 
-    GridBlock(int xStart, int yStart, Block myBlock)
-    {
+    Block block;
+
+    Grid grid;
 
+    public GridBlock(int xStart, int yStart, Block myBlock)
+    {
+        x = xStart;
+        y = xStart;
+        block = myBlock;
     }
 
+    //Return the x index of the gridblock
     public int GetX()
     {
         return x;
     }
 
+    //Return the y index of the gridblock
     public int GetY()
     {
         return y;
     }
 
+    /*Enable gridblocks to move according to their directions.
+     Gridblock moves if it can fit into the next position.The 
+     moving process is not physical. Instead, it duplicates
+     all the tiles to the next position and erases the original
+     ones.*/
     public void Move(Enums.Direction direction)
     {
+        int tempX = x;
+        int tempY = y;
+        int blockW = block.GetWidth();
+        int blockH = block.GetHeight();
+        Tile[,] gridTiles = grid.GetTiles();
+
+        switch (direction)
+        {
+            case Enums.Direction.Down:
+                y += blockH;
+                break;
+            case Enums.Direction.Up:
+                y -= blockH;
+                break;
+            case Enums.Direction.Left:
+                x -= blockW;
+                break;
+            case Enums.Direction.Right:
+                x += blockW;
+                break;
+        }
 
+        if (grid.CanBlockFit(x, y, block))
```

Comment:
On second thought, CanBlockFit isn't the ideal function to call here, since it doesn't exempt the Tiles of this Block itself. My bad - I have updated GridBlock's entry in the TDD with a better idea to approach this.